### DISCUSSION TITLE
Can't Get Click To work?

### DISCUSSION THREAD

**Discussion by glass22** [spyoungtech/ahk#304] [Q&A] [answered]
I'm trying to make a click without mouse moving, can't get it to work at all?

Simple code

```
from ahk import AHK
import time

myK = AHK()

time.sleep(2)
win = myK.active_window

a=0

while a<3:
    time.sleep(1)
    print(myK.get_mouse_position())
    a+=1
    win.click(myK.get_mouse_position())
```

Also what paramaters does control_click() need?

**Comment by spyoungtech** [comment, marked as answer]
`win.click` expects separate x,y arguments. It's not quite consistent with the regular `click` function, which also accepts a tuple in place of the `x` argument.

You could make a small adjustment and use `*` unpacking to do this:
```diff
- win.click(myK.get_mouse_position())
+ win.click(*myK.get_mouse_position())
```

Or in the latest version of the library, you can also do something like this:

```py
pos = myK.get_mouse_position()
win.click(pos.x, pos.y)
```

> Also what paramaters does control_click() need?

You can always look at the signature in the code or in the [api documentation](https://ahk.readthedocs.io/en/latest/api/index.html):

- [control_click](https://ahk.readthedocs.io/en/latest/api/sync.html#ahk._sync.engine.AHK.control_click)
- [Window.click](https://ahk.readthedocs.io/en/latest/api/sync.html#ahk._sync.window.Window.click) (which calls control_click for you).

The meanings of the arguments should either be plain or the same as in the AutoHotkey documentation .
